command/base.go

```diff
@@ -393,6 +394,13 @@ func (c *BaseCommand) flagSet(bit FlagSetBit) *FlagSets {
 					"This can be specified multiple times.",
 			})
 
+			f.BoolVar(&BoolVar{
+				Name:    "non-interactive",
+				Target:  &c.flagNonInteractive,
+				Default: false,
+				Usage:   "It controls a command to be executed in an interactive or non-interactive fashion with a user.",
+			})
+
 		}
 
```

Comment:
Can we also use [github.com/mattn/go-isatty](http://github.com/mattn/go-isatty) to override the var to true when we *can't* ask the user for input (because there's no tty)?

Reply:
Actually, ui.AskSecret already takes care of that here:
https://github.com/mitchellh/cli/blob/5454ffe87bc5c6d8b6b21c825617755e18a07828/ui.go#L79

Reply:
I think this is less desirable, because then if the user didn't say -non-interactive, but they're running without a terminal, we'll simply block waiting for input on stdin.  This will be a surprising change and it won't be obvious what the problem is.